File: qa_exports/capture_all_formats.py
"""
Playwright capture — all 7 FIFA Classic export formats
Usage: python qa_exports/capture_all_formats.py
"""
import sys
from pathlib import Path
from playwright.sync_api import sync_playwright

BASE_URL = "http://localhost:8000"
USER_ID  = 19310
OUT_DIR  = Path(__file__).parent

FORMATS = [
    ("portrait",  "instagram_portrait",  1080, 1350),
    ("story",     "instagram_story",     1080, 1920),
    ("square",    "instagram_square",    1080, 1080),
    ("landscape", "facebook_landscape",  1200,  630),
    ("tiktok",    "tiktok",             1080, 1920),
    ("banner",    "banner_custom",       1500,  500),
    ("og",        "og",                 1200,  630),
]

results = []

with sync_playwright() as pw:
    browser = pw.chromium.launch(headless=True)

    for bucket, platform, w, h in FORMATS:
        url = f"{BASE_URL}/players/{USER_ID}/card?platform={platform}&export=1"
        out = OUT_DIR / f"{bucket}_{platform}.png"

        page = browser.new_page()
        page.set_viewport_size({"width": w, "height": h})
        resp = page.goto(url, wait_until="networkidle", timeout=15_000)
        status = resp.status
        # Let font-display:block fonts paint
        page.wait_for_timeout(800)
        page.screenshot(path=str(out), clip={"x": 0, "y": 0, "width": w, "height": h})
        page.close()

        size_kb = out.stat().st_size / 1024
        ok = status == 200 and size_kb > 10
        results.append((bucket, platform, status, out.name, size_kb, ok))
        icon = "✅" if ok else "❌"
        print(f"{icon}  {bucket:<10}  HTTP {status}  →  {out.name}  ({size_kb:.1f} KB)")

    browser.close()

print("\n── Summary ──")
all_ok = all(r[5] for r in results)
print("✅  All 7 exports OK" if all_ok else "❌  Some exports failed")
if not all_ok:
    for r in results:
        if not r[5]:
            print(f"   FAIL: {r[0]} — HTTP {r[2]}, {r[3]:.1f} KB")
    sys.exit(1)

File: qa_exports/capture_all_formats.js
/**
 * Playwright capture script — all 7 FIFA Classic export formats
 * Usage: node qa_exports/capture_all_formats.js
 * Requires: npx playwright install chromium (once)
 */
const { chromium } = require('playwright');
const path = require('path');
const fs = require('fs');

const BASE_URL = 'http://localhost:8000';
const USER_ID  = 19310;
const OUT_DIR  = path.join(__dirname);

const FORMATS = [
  { bucket: 'portrait',  platform: 'instagram_portrait',  w: 1080, h: 1350 },
  { bucket: 'story',     platform: 'instagram_story',     w: 1080, h: 1920 },
  { bucket: 'square',    platform: 'instagram_square',    w: 1080, h: 1080 },
  { bucket: 'landscape', platform: 'facebook_landscape',  w: 1200, h: 630  },
  { bucket: 'tiktok',    platform: 'tiktok',              w: 1080, h: 1920 },
  { bucket: 'banner',    platform: 'banner_custom',       w: 1500, h: 500  },
  { bucket: 'og',        platform: 'og',                  w: 1200, h: 630  },
];

(async () => {
  const browser = await chromium.launch({ headless: true });
  const results = [];

  for (const fmt of FORMATS) {
    const url = `${BASE_URL}/players/${USER_ID}/card?platform=${fmt.platform}&export=1`;
    const outFile = path.join(OUT_DIR, `${fmt.bucket}_${fmt.platform}.png`);

    const page = await browser.newPage();
    await page.setViewportSize({ width: fmt.w, height: fmt.h });
    const resp = await page.goto(url, { waitUntil: 'networkidle', timeout: 15000 });
    const status = resp.status();

    // Wait for fonts (font-display:block) to render
    await page.waitForTimeout(800);

    await page.screenshot({ path: outFile, fullPage: false, clip: { x: 0, y: 0, width: fmt.w, height: fmt.h } });
    await page.close();

    const size = fs.statSync(outFile).size;
    results.push({ bucket: fmt.bucket, platform: fmt.platform, status, file: path.basename(outFile), bytes: size });
    console.log(`✅  ${fmt.bucket.padEnd(10)} HTTP ${status}  →  ${path.basename(outFile)}  (${(size/1024).toFixed(1)} KB)`);
  }

  await browser.close();

  console.log('\n── Summary ──');
  const allOk = results.every(r => r.status === 200 && r.bytes > 10_000);
  console.log(allOk ? '✅  All 7 exports OK' : '❌  Some exports failed');
  if (!allOk) {
    results.filter(r => r.status !== 200 || r.bytes <= 10_000)
           .forEach(r => console.error(`  FAIL: ${r.bucket} — HTTP ${r.status}, ${r.bytes} bytes`));
    process.exit(1);
  }
})();
